pennylane/data/utils.py

```diff
@@ -14,311 +14,181 @@
+        )
 
-    .. code-block :: pycon
+    if not isinstance(attributes, list):
+        raise TypeError(f"Arg 'attributes' should be a list, but got {type(attributes)}.")
 
-        >>> _convert_size(1024)
-        1 KB
+    all_attributes = data["attributes"]
+    if not set(attributes).issubset(set(all_attributes)):
+        raise ValueError(
+            f"Supported key values for {data_type} are {all_attributes}, but got {attributes}."
+        )
 
-    """
 
-    if not size_bytes:
-        return "0 B"
-    size_name = ("B", "KB", "MB", "GB", "TB", "PB", "EB", "ZB", "YB")
-    indx = int(math.floor(math.log(size_bytes, 1024)))
-    size = round(size_bytes / math.pow(1024, indx), 2)
-    return f"{size} {size_name[indx]}"
+def _refresh_foldermap():
+    """Refresh the foldermap from S3."""
+    response = requests.get(FOLDERMAP_URL, timeout=5.0)
+    if not response.ok:
+        response.raise_for_status()
 
+    global _foldermap
+    _foldermap = response.json()
 
-def _write_prog_bar(progress, completed, barsize, barlength, total_length):
-    r"""Helper function for printing progress bar for downloads
 
-    Args:
-        progress (float): File size in bytes of the file currently being downloaded
-        completed (float): Bar size representing the file currently being downloaded
-        barsize (float): Bar size representing the download bar
-        barlength (float): Length of the bar printed for showing downloading progress
-        total_length (float): Total size in bytes of the file currently being downloaded
+def _refresh_data_struct():
+    """Refresh the data struct from S3."""
+    response = requests.get(DATA_STRUCT_URL, timeout=5.0)
+    if not response.ok:
+        response.raise_for_status()
 
-    Returns:
-        Prints the progressbar to the console
-    """
-    f = f"[{chr(9608)*barlength} {round(completed, 3)} %{'.'*(barsize-barlength)}] {_convert_size(progress)}/{_convert_size(total_length)}"
-    sys.stdout.write("\r" + f)
-    sys.stdout.flush()
+    global _data_struct
+    _data_struct = response.json()
 
 
-def _validate_params(data_type, description, attributes=None):
-    r"""Validate parameters for loading the data"""
+def _fetch_and_save(filename, dest_folder):
+    """Download a single file from S3 and save it locally."""
+    response = requests.get(os.path.join(S3_URL, filename), timeout=5.0)
+    if not response.ok:
+        response.raise_for_status()
+    with open(os.path.join(dest_folder, filename), "wb") as f:
+        f.write(response.content)
 
-    if data_type not in list(DATA_STRUCT.keys()):
-        raise ValueError(
-            f"Currently we have data hosted from types: qchem and qspin, but got {data_type}."
-        )
 
-    if not isinstance(description, dict):
-        raise TypeError(f"Args 'description' should be a dict, but got {type(description)}.")
+def _s3_download(data_type, folders, attributes, dest_folder, force, num_threads):
```

Comment:
Should the default values of keywords be added here as well?

Reply:
I didn't because it's a protected method that's always called from within source code, and that code always provides all these arguments. Default values just ask for trouble imo